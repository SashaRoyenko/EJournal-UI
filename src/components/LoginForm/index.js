import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../shared/svg/school.svg';
import {
  FormLogo,
  FormInput,
  FormButton,
  FormMessage,
  FormContainer,
} from '../../shared/components/Form/styled';

export default class LoginForm extends React.Component {
  render() {
    const { onClickMessageHandler } = this.props;
    return (
      <FormContainer>
        <FormLogo src={logo} alt="logo" />
        <form>
          <FormInput type="text" placeholder="username" />
          <FormInput type="password" placeholder="password" />
          <FormButton>login</FormButton>
          <FormMessage>
            <span>Already registered? </span>
            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
            <a href="#" onClick={onClickMessageHandler}>Sign In</a>
          </FormMessage>
        </form>
      </FormContainer>
    );
  }
}

LoginForm.propTypes = {
  onClickMessageHandler: PropTypes.func,
};

LoginForm.defaultProps = {
  onClickMessageHandler: () => {},
};
