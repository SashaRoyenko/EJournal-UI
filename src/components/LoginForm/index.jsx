import React from 'react';
import logo from '../../shared/svg/school.svg';
import {
    FormLogo,
    FormInput,
    FormButton,
    FormMessage,
    FormContainer,
} from '../../shared/components/Form/styled.js';

export default class LoginForm extends React.Component {
    
    render() {
        return (
            <FormContainer>
                <FormLogo src={logo} alt="logo" />
                <form>
                    <FormInput type="text" placeholder="username" />
                    <FormInput type="password" placeholder='password' />
                    <FormButton>login</FormButton>
                    <FormMessage>
                        <span>Already registered? </span>
                        <a href="#" onClick={this.props.onClickMessageHandler}>Sign In</a>
                    </FormMessage>
                </form>
            </FormContainer>
        );
    }
}
