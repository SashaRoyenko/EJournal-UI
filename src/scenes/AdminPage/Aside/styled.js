import styled from 'styled-components';
import { Link } from "react-router-dom";
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  overflow: hidden;
  margin-right: 20px;
  border-radius: 5px;
  width: 190px;
  height: calc(100vh - 215px);
  background-color: ${cssConst.themeMedium};
`;

export const MenuItem = styled(Link)`
  display: block;
  border-bottom: solid 1px ${cssConst.themeLight};
  padding: 10px 15px;
  font-size: 16px;
  font-family: Roboto;
  text-decoration: none;
  color: ${cssConst.labelHighlight};

  &:active {
    background: ${cssConst.themeLightMedium};
  }
`;
