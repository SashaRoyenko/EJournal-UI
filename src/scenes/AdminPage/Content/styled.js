import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  width: calc(100% - 210px);
  height: calc(100vh - 215px);
  padding: 12px 15px 20px;
`;

export const Header = styled.p`
  margin-bottom: 10px;
  font-size: 28px;
  font-family: Roboto;
  font-weight: bold;
  color: ${cssConst.themeDark};
`;
