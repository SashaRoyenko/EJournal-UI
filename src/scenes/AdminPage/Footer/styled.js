import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 20px;
  box-shadow: 0 8px 6px 0 ${cssConst.lightShadow};
  border-radius: 5px;
  width: 100%;
  height: 50px;
  padding: 15px;
  background-color: ${cssConst.themeDark};
`;

export const Label = styled.span`
  font-size: 14px;
  font-family: Roboto;
  color: ${cssConst.labelHighlight};
`;