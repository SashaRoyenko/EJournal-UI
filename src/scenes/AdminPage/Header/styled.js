import styled from 'styled-components';
import { Link } from "react-router-dom";
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 15px;
  box-shadow: 0 8px 6px 0 ${cssConst.lightShadow};
  border-radius: 5px;
  width: 100%;
  height: 100px;
  padding: 15px;
  background-color: ${cssConst.themeDark};
`;

export const HomeLink = styled(Link)`
  font-size: 34px;
  font-family: Roboto;
  text-decoration: none;
  color: ${cssConst.labelHighlight};
`;

export const LoginWrapper = styled.div`
  width: 240px;
`;

export const Label = styled.span`
  font-size: 16px;
  font-family: Roboto;
  font-weight: ${(p) => (p.highlighted ? 'bold' : 'normal')};
  text-decoration: ${(p) => (p.highlighted ? 'underline' : 'none')};
  color: ${(p) => (p.highlighted ? cssConst.labelHighlight : cssConst.labelDefault)};
`;

export const LogOutButton = styled.button`
  margin: 4px 0 0;
  border: none;
  padding: 0;
  background: none;
  font-size: 16px;
  font-family: Roboto;
  color: ${cssConst.labelHighlight};

  &:hover{
    text-decoration: underline;
  }
`;
