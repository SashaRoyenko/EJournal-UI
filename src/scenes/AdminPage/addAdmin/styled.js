import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  width: calc(100% - 210px);
  height: calc(100vh - 215px);
  padding: 12px 15px 20px 0px;
  border-radius: 5px;
`;

export const ItemTitle = styled.p`
  padding-bottom:10px;
`;

export const TableItem = styled.td`
  padding: 10px 30px 34px 0px;
  margin-bottom: 10px;
  font-size: 16px;
  font-family: Roboto;
  font-weight: bold;
  color: ${cssConst.themeDark};
`;

export const Input = styled.input`
  width: 275px;
  height: 25px;
  border-radius: 3px;
  border: solid 1px #495057;
  background-color: #ffffff;
`;

export const Button = styled.button`
  width: 100px;
  height: 25px;
  border-radius: 7px;
  font-family: Roboto;
  background-color: ${cssConst.buttonBlue};
  font-size: 12px;
  color: white;
  font-weight: bold;
  border:none;
  
  &:active{
  background-color: ${cssConst.buttonBlueOnClick};
  border: 2px solid ${cssConst.turquoise};
  }
`;
