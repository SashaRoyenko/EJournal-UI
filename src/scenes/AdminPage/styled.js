import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 15px;
`;

export const Main = styled.div`
  display: flex;
`;
