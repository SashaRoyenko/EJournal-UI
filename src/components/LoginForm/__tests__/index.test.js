import React from 'react';
import { mount } from 'enzyme';
import { FormMessage } from '../../../shared/components/Form/styled';
import LoginForm from '../index';

describe('<LoginForm />', () => {
  const onClickMessageHandlerMock = jest.fn();
  const mandatoryProps = {
    onClickMessageHandler: onClickMessageHandlerMock,
  };

  it('should render correctly with default props', () => {
    const sut = mount(<LoginForm />);

    expect(sut).toMatchSnapshot();

    sut.find(FormMessage).find('a').simulate('click');
  });

  it('should render correctly', () => {
    const sut = mount(<LoginForm {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should call handler on click on redirection link', () => {
    const sut = mount(<LoginForm {...mandatoryProps} />);

    expect(onClickMessageHandlerMock).toBeCalledTimes(0);

    sut.find(FormMessage).find('a').simulate('click');

    expect(onClickMessageHandlerMock).toBeCalledTimes(1);
  });
});
