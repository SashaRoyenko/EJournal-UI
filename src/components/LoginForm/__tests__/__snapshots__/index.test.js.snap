// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render correctly 1`] = `
<LoginForm
  onClickMessageHandler={[MockFunction]}
>
  <styled.div>
    <div
      className="sc-AxjAm kFqgcy"
    >
      <styled.img
        alt="logo"
        src={Object {}}
      >
        <img
          alt="logo"
          className="sc-AxhUy fdjSkp"
          src={Object {}}
        />
      </styled.img>
      <form>
        <styled.input
          placeholder="username"
          type="text"
        >
          <input
            className="sc-AxirZ PTCQu"
            placeholder="username"
            type="text"
          />
        </styled.input>
        <styled.input
          placeholder="password"
          type="password"
        >
          <input
            className="sc-AxirZ PTCQu"
            placeholder="password"
            type="password"
          />
        </styled.input>
        <styled.button>
          <button
            className="sc-AxiKw bAyasi"
          >
            login
          </button>
        </styled.button>
        <styled.p>
          <p
            className="sc-AxhCb eVMfSZ"
          >
            <span>
              Already registered? 
            </span>
            <a
              href="#"
              onClick={[MockFunction]}
            >
              Sign In
            </a>
          </p>
        </styled.p>
      </form>
    </div>
  </styled.div>
</LoginForm>
`;

exports[`<LoginForm /> should render correctly with default props 1`] = `
<LoginForm
  onClickMessageHandler={[Function]}
>
  <styled.div>
    <div
      className="sc-AxjAm kFqgcy"
    >
      <styled.img
        alt="logo"
        src={Object {}}
      >
        <img
          alt="logo"
          className="sc-AxhUy fdjSkp"
          src={Object {}}
        />
      </styled.img>
      <form>
        <styled.input
          placeholder="username"
          type="text"
        >
          <input
            className="sc-AxirZ PTCQu"
            placeholder="username"
            type="text"
          />
        </styled.input>
        <styled.input
          placeholder="password"
          type="password"
        >
          <input
            className="sc-AxirZ PTCQu"
            placeholder="password"
            type="password"
          />
        </styled.input>
        <styled.button>
          <button
            className="sc-AxiKw bAyasi"
          >
            login
          </button>
        </styled.button>
        <styled.p>
          <p
            className="sc-AxhCb eVMfSZ"
          >
            <span>
              Already registered? 
            </span>
            <a
              href="#"
              onClick={[Function]}
            >
              Sign In
            </a>
          </p>
        </styled.p>
      </form>
    </div>
  </styled.div>
</LoginForm>
`;
